x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -283,6 +337,9 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeGenericMap(transientMetadata);
         out.writeArray(ApplicationResourcePrivileges::write, applicationPrivileges);
         ConfigurableClusterPrivileges.writeArray(out, getConditionalClusterPrivileges());
+        if (out.getVersion().onOrAfter(Version.V_8_6_0)) {
+            out.writeOptionalArray(remoteIndicesPrivileges);
+        }
```

Comment:
I suggest we throw error here if it has any remote indices and the wire version is earlier instead of silently skipping the field.

Request and reponse objects have rather clear code path. We can reliably know that some of them are never sent across wire. Hence how `writeTo` method behaves does not make much difference for them. But entity object like this one is harder to know for sure. It might not be sent across wire today. But its nature does not prevent it from being used that way.

Reply:
I've addressed this. Will do the same for `PutRoleRequest`